### PR TITLE
Allow VirtualTimeScheduler to run on any thread 

### PR DESCRIPTION
**Pull request by danielt1263**
…as long as critical methods are all called on the same thread.

This is designed to fix issue #2609

### CODE REVIEW

**Comment by danielt1263** [comment]
As far as I can tell, this is 100% backwards compatible and doesn't require any API change from users of the library. It may not solve every future use-case that Swift Testing might throw at it, but it solves the single most obvious one.

I see no reason that this can't be accepted into the next minor release/update.

File: RxSwift/Schedulers/VirtualTimeScheduler.swift [hunk 1]
```diff
@@ -106,7 +108,10 @@ open class VirtualTimeScheduler<Converter: VirtualTimeConverterType>
      - returns: The disposable object used to cancel the scheduled action (best effort).
      */
     public func scheduleAbsoluteVirtual<StateType>(_ state: StateType, time: VirtualTime, action: @escaping (StateType) -> Disposable) -> Disposable {
-        MainScheduler.ensureExecutingOnScheduler()
+        guard thread == nil || Thread.current == thread else {
+            rxFatalError("Executing on the wrong thread. Please ensure all work on the same thread.")
+        }
+        thread = Thread.current
```
Comment (nikolaykasyanov) [inline RxSwift/Schedulers/VirtualTimeScheduler.swift:114]: Is it safe to read and write `thread` without any locking?

Comment (danielt1263) [reply RxSwift/Schedulers/VirtualTimeScheduler.swift:114]: Isn't that what we always do on the main thread? Read/write without locking? Maybe I don't understand the question...

Comment (nikolaykasyanov) [reply RxSwift/Schedulers/VirtualTimeScheduler.swift:114]: Indeed, it should be completely fine as long scheduler's methods are actually called on one thread, but what if there would be two close calls from different threads, it's not impossible that both will see `thread == nil` and proceed, for example.

Comment (danielt1263) [reply RxSwift/Schedulers/VirtualTimeScheduler.swift:114]: Right! Good point. I will update to account for that.

Comment (nikolaykasyanov) [reply RxSwift/Schedulers/VirtualTimeScheduler.swift:114]: @danielt1263 sounds good!

**Comment by bennnjamin** [comment]
What's the hold up in getting this merged? Would love to start being able to migrate our RxSwift tests to Swift Testing

**Comment by freak4pc** [comment]
We'll need someone with a meaningful RxSwift-based test suite to test this to make sure it doesn't break existing tests.

**Comment by nikolaykasyanov** [comment]
@freak4pc I'll try it out and report back.

**Comment by freak4pc** [comment]
> @freak4pc I tried this PR on our relatively big test suite (around 1800 tests, most of which use `TestScheduler`) and didn't find any issues ✅ . To be sure, I also ran the suite repeatedly 10 times ✅ and with thread sanitizer ✅ .

That's awesome, thanks Nikolay!

**Pull request merged**